src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -231,7 +231,7 @@ private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
                 // TODO: Remove this when implicit operations are synthesised: https://github.com/dotnet/roslyn/issues/47829 
                 var methodSyntax = method.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
                 if (method.IsConstructor() &&
-                    _compilationAnalyzer.IsRecordDeclaration(methodSyntax))
+                    (_compilationAnalyzer.IsRecordDeclaration(methodSyntax) || _compilationAnalyzer.IsClassDeclaration(methodSyntax)))
```

Comment:
Should this just be `_compilationAnalyzer.IsTypeDeclaration(methodSyntax)` and replace both methods with a single `IsTypeDeclaration` method?